pkg/kv/kvserver/replica_application_result.go

```diff
@@ -407,6 +407,8 @@ func (r *Replica) tryReproposeWithNewLeaseIndex(ctx context.Context, origCmd *re
 		// The tracker wants us to forward the request timestamp, but we can't
 		// do that without re-evaluating, so give up. The error returned here
 		// will go to back to DistSender, so send something it can digest.
+		r.mu.RLock()
```

Comment:
The lock seems legit. At least, the `r.mu.proposalBuf` access above is assumed without holding the lock (see `prpposalBuf` comment), and `TrackEvaluatingRequest` also locks/unlocks the `r.mu.RLock()`.

We're only holding `raftMu` here (in the weeds of application flow), and it's ok to lock `r.mu` after it.